django/views/defaults.py

```diff
@@ -59,12 +74,19 @@ def server_error(request, template_name='500.html'):
     try:
         template = loader.get_template(template_name)
     except TemplateDoesNotExist:
+        if template_name != ERROR_500_TEMPLATE_NAME:
+            warnings.warn(
+                "A template_name has been provided, but the template cannot be found. "
```

Comment:
@yakky Sorry, missed your comments somehow. I was thinking the two pieces of information that would be useful in these messages would be a) the name of the view function and b) the name of the template. (b) is more critical; without (b) these deprecation warnings are even less useful than the raw `TemplateDoesNotExist`.

But @timgraham has raised the question of whether we really need a deprecation path here at all, which would make it a moot point, so that question should be resolved first.